### ISSUE TITLE
Offers to enrol with roles that can't be assigned  on the course level

### ISSUE THREAD

**Issue by mudrd8mz** [at-tools/moodle-block_uploadgroups#3]
Role definition contains a list of contexts where the role can be assigned. I believe that only roles that can be assigned on the Course context level should be offered (if some should be offered at all - see my reasoning in issue #2 where I suggest to get rid of this functionality completely).

**Comment by mudrd8mz** [comment]
![screenshot_2016-01-14_16-27-17](https://cloud.githubusercontent.com/assets/85831/12328434/b503978c-badb-11e5-8dc1-0b89f1dcb6f2.png)

**Comment by cdsmith-umn** [comment]
@mudrd8mz You're absolutely right that this should respect the Course context permissions/restrictions/settings, assuming it's retained. Little else to say about this, as #2 does indeed cover this and more.

**Comment by cdsmith-umn** [comment]
@mudrd8mz I was thinking about the wrong thing when I wrote my last comment. I thought this was specifically saying something about roles the user can assign. Could you explain what it is you're talking about here? Also, could you give me an idea where I could go to look at the page you've screenshotted? Thank you.

**Comment by mudrd8mz** [comment]
This was screenshot'ed from the standard role definition page.
